### PR TITLE
[RTM]Fmriprep issue308

### PR DESCRIPTION
**Pull request by rwblair**

### CODE REVIEW

**Comment by rwblair** [comment]
fixes in part both poldracklab/fmriprep#307 and poldracklab/fmriprep#308

File: niworkflows/viz/utils.py [hunk 1]
```diff
@@ -65,19 +66,24 @@ def save_html(template, report_file_name, unique_string, **kwargs):
 
 
 def as_svg(image, filename='temp.svg'):
-    ''' takes an image as created by nilearn.plotting and returns a blob svg.
-    A bit hacky. '''
+    ''' Takes an image as created by nilearn.plotting and returns the svg with
+        extra headers removed and the width and height style values set to
+        nothing. '''
     image.savefig(filename)
     with open(filename, 'r' if PY3 else 'rb') as file_obj:
```
Comment (oesteban) [inline niworkflows/viz/utils.py]: We should replace writing svgs to the filesystem, in favor of something like https://github.com/oesteban/mriqc/blob/master/mriqc/viz/svg.py

Maybe for another PR?

File: niworkflows/viz/utils.py [hunk 2]
```diff
@@ -65,19 +66,24 @@ def save_html(template, report_file_name, unique_string, **kwargs):
...
         image_svg = file_obj.readlines()
 
-    svg_start = 0
+    svg_start = None
+    svg_lines_corrected = []
     for i, line in enumerate(image_svg):
```
Comment (oesteban) [inline niworkflows/viz/utils.py]: Particularly, I find it more readable if we did this this way: https://github.com/oesteban/mriqc/blob/master/mriqc/viz/svg.py#L63

**Comment by rwblair** [comment]
Trying out changes recommended by @oesteban

File: niworkflows/viz/utils.py [hunk 3]
```diff
@@ -256,11 +303,17 @@ def compose_view(bg_svgs, fg_svgs, ref=0, out_file='report.svg'):
...
+
     # Query the size of each
-    sizes = np.array([(int(f.width[:-2]), int(f.height[:-2])) for f in svgs])
+    sizes = []
+    for f in svgs:
+        viewbox = f.root.get("viewBox").split(" ")
+        width = int(viewbox[2])
+        height = int(viewbox[3])
+        sizes.append((width, height))
+    nsvgs = len(bg_svgs)
```
Comment (effigies) [inline niworkflows/viz/utils.py]: `sizes` is now expected to be an `ndarray`.

`sizes = np.array(sizes)` here should fix things.

**Comment by rwblair** [comment]
Final integration results here:
https://circleci.com/gh/rwblair/fmriprep/54

File: niworkflows/viz/utils.py [hunk 4]
```diff
@@ -64,22 +66,38 @@ def save_html(template, report_file_name, unique_string, **kwargs):
...
+def svg2str(display_object, dpi=300):
+    """
+    Serializes a nilearn display object as a string
+    """
+    from io import StringIO
+    image_buf = StringIO()
+    display_object.frame_axes.figure.savefig(
+        image_buf, dpi=dpi, format='svg',
+        facecolor='k', edgecolor='k')
+    image_buf.seek(0)
```
Comment (effigies) [inline niworkflows/viz/utils.py]: No need to seek in a `StringIO`. `getvalue()` always returns the entire buffer.

File: niworkflows/viz/utils.py [hunk 5]
```diff
@@ -234,17 +252,22 @@ def plot_registration(anat_nii, div_id, plot_params=None,
...
+            NIWORKFLOWS_LOG.info(e)
+            return
         find_text = etree.ETXPath("//{%s}g[@id='figure_1']" % (SVGNS))
         find_text(xml_data)[0].set('id', '%s-%s-%s' % (div_id, mode, uuid4()))
 
-        with open(out_file, 'wb') as f:
-            f.write(etree.tostring(xml_data))
+        out_files.append(etree.tostring(xml_data))
+        #with open(out_file, 'wb') as f:
+        #    f.write(etree.tostring(xml_data))
```
Comment (effigies) [inline niworkflows/viz/utils.py]: Do the comments need to be here?

File: niworkflows/viz/utils.py (hunk 3 above)
Comment (effigies) [inline niworkflows/viz/utils.py]: If you want to be really fancy and define sizes in one go:

```Python
sizes = np.array([tuple(map(int, f.root.get("viewBox").split(" ")[2:4]))
                  for f in svgs])
# or
sizes = np.array([tuple(int(dim) for dim in f.root.get("viewBox").split(" ")[2:4])
                  for f in svgs])
```

**Pull request merged**